src/app/compiler/compiler-input.js

```diff
@@ -0,0 +1,25 @@
+'use strict'
+
+/*
+  opts:
+   - optimize
+   - { file_name: { library_name: address } }
+*/
+module.exports = (sources, opts) => {
+  return JSON.stringify({
+    language: 'Solidity',
+    sources,
```

Comment:
I think this should be `sources: sources` ?

Reply:
`sources, ` like this also work

Reply:
What does it do? What is `sources`? Is it `{ sources: {} }`?

Reply:
sources is a map of sources.
that's just a shorthand for `sources: sources,`

Reply:
I think the problem with this construct is that not many are familiar with it (at least I aren't) and that if someone goes to refactor the variable name, it will break the JSON.